lib/dataspec/src/lib.rs

```diff
@@ -1,7 +1,19 @@
 //! Format of JSON output from rpc-perf. These structures can be used
 //! by any consumer of the produced data to parse the files.
+use datafusion::arrow::array::*;
+use datafusion::arrow::datatypes::*;
+use datafusion::parquet::arrow::ArrowWriter;
+use datafusion::parquet::basic::{Compression, ZstdLevel};
+use datafusion::parquet::errors::ParquetError;
+use datafusion::parquet::file::properties::WriterProperties;
+use datafusion::parquet::format::FileMetaData;
 use histogram::SparseHistogram;
 use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use std::fs::File;
+use std::sync::Arc;
+
+const PERCENTILES: &[&str] = &["p25", "p50", "p75", "p90", "p99", "p999", "p9999", "max"];
```

Comment:
Should we just embed the full histogram instead of having predefined percentiles?

Reply:
We do embed the full histogram too. I think Xi added the summary statistics so that it allowed us to scan for points of interest without parsing the full histogram.